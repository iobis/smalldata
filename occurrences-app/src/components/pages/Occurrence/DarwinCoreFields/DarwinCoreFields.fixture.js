import DarwinCoreFields from './DarwinCoreFields'

export default [{
  component: DarwinCoreFields,
  name:      'DarwinCoreFields',
  props:     {
    fields:   getDefaultFields(),
    onChange: console.log
  }
}]

export function getDefaultFields() {
  return [
    { name: 'name-1', value: 'value-1' },
    { name: 'name-2', value: 'value-2' },
    { name: 'name-3', value: 'value-3' }
  ]
}
