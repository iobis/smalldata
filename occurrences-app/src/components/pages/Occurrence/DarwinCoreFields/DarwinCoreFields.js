import PropTypes from 'prop-types'
import React, {useState} from 'react'
import InputText from '../../../form/InputText'

export default function DarwinCoreFields({ onChange }) {

  const [fields,setFields] = useState([
    { name: 'dummy field', value: 'dummy value' },
    { name:  'dummy2 field', value: 'dummy value' },
    { name: 'dummy3 field', value: 'dummy value' },
    { name: 'dummy4 field', value: 'dummy value' }
  ])

  const [name, setName] = useState('')
  const [value, setValue] = useState('')


  function makeDarwinCoreObject(){
    setFields([...fields,{name,value}])
  }

  return (
    <div className="darwin-core-fields section is-fluid">

      <h2>DARWIN CORE CUSTOM SELECTION</h2>
      <p>This form enables you to add any aditional fields you may need to specify, that were ot previously included in
        this form.
        Please be advised to use the Darwin Core Archive names</p>

      <div className="columns">
        <InputText className="field-name" name="occurrenceForm.darwinCoreFields.fieldName" onChange={setName}/>
        <InputText className="value" name="occurrenceForm.darwinCoreFields.value" onChange={setValue}/>
        <div className="column add"><a className="button" onClick={makeDarwinCoreObject}>add</a></div>
      </div>

      <table>
        <thead>
        <tr>
          <th>name</th>
          <th>value</th>
        </tr>
        </thead>
        <tbody>
        {fields.map(field =>
          <tr className="fieldrow" key={field.name + field.value}>
            <td>{field.name}</td>
            <td>{field.value}</td>
            
          </tr>
        )}
        </tbody>
      </table>

    </div>
  )
}

DarwinCoreFields.propTypes = {
  onChange: PropTypes.func.isRequired
}
