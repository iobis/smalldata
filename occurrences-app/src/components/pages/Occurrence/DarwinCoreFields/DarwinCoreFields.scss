.darwin-core-fields
 {
  /*.column {
    background-color: green
  }*/
  .add{
    display: flex;
   /* background-color: yellow;*/
  }
  .button {
    align-self: center;
  }
}

