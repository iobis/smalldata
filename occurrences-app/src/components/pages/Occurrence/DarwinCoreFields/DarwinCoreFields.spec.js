import React from 'react'
import DarwinCoreFields from './DarwinCoreFields'
import { mount } from 'enzyme'
import { getDefaultFields } from './DarwinCoreFields.fixture'

describe('DarwinCoreFields', () => {
  let wrapper

  it('renders correctly', () => {
    expect(mount(createComponent())).toMatchSnapshot()
  })

  it('adds element when clicking add button', () => {
    const onChange = jest.fn()
    const wrapper = mount(createComponent({ onChange }))
    expect(wrapper.find('.fieldrow')).toHaveLength(3)

    wrapper.find('.field-name input').simulate('change', { target: { value: 'new name' } })
    wrapper.find('.value input').simulate('change', { target: { value: 'new value' } })
    wrapper.find('.add .button').simulate('click')
    expect(wrapper.find('.fieldrow')).toHaveLength(4)
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toBeCalledWith([
      ...getDefaultFields(),
      { 'name': 'new name', 'value': 'new value' }
    ])
  })

  it('removes element when clicking remove button', () => {
    const onChange = jest.fn()
    wrapper = mount(createComponent({ onChange }))

    wrapper.find('.remove').first().simulate('click')
    expect(wrapper.find('.fieldrow')).toHaveLength(2)
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toBeCalledWith([getDefaultFields()[1], getDefaultFields()[2]])
  })

  it('updates values when updates props', () => {
    wrapper = mount(createComponent({ fields: [] }))
    expect(wrapper.find('.fieldrow')).toHaveLength(0)

    wrapper.setProps({ fields: getDefaultFields() })
    wrapper.update()
    expect(wrapper.find('.fieldrow')).toHaveLength(3)
  })
})

function createComponent(props) {
  const defaultProps = {
    fields:   getDefaultFields(),
    onChange: jest.fn()
  }

  return <DarwinCoreFields {...defaultProps} {...props}/>
}
