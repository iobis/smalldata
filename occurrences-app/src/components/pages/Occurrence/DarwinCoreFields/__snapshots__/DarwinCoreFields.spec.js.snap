// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarwinCoreFields renders correctly 1`] = `
<DarwinCoreFields
  onChange={[MockFunction]}
>
  <div
    className="darwin-core-fields section is-fluid"
  >
    <h2>
      DARWIN CORE CUSTOM SELECTION
    </h2>
    <p>
      This form enables you to add any aditional fields you may need to specify, that were ot previously included in this form. Please be advised to use the Darwin Core Archive names
    </p>
    <div
      className="columns is-grouped"
    >
      <InputText
        className="field-name"
        name="occurrenceForm.darwinCoreFields.fieldName"
        onChange={[Function]}
      >
        <div
          className="column field field-name"
        >
          <label
            className="label"
          >
            occurrenceForm.darwinCoreFields.fieldName.label
          </label>
          <input
            className="input"
            onChange={[Function]}
            type="text"
          />
        </div>
      </InputText>
      <InputText
        className="value"
        name="occurrenceForm.darwinCoreFields.value"
        onChange={[Function]}
      >
        <div
          className="column field value"
        >
          <label
            className="label"
          >
            occurrenceForm.darwinCoreFields.value.label
          </label>
          <input
            className="input"
            onChange={[Function]}
            type="text"
          />
        </div>
      </InputText>
      <div
        className="column add"
      >
        <a
          className="button"
          onClick={[Function]}
        >
          add
        </a>
      </div>
    </div>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            name
          </th>
          <th>
            value
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr
          className="fieldrow"
          key="dummy fielddummy value"
        >
          <td>
            dummy field
          </td>
          <td>
            dummy value
          </td>
          <td>
            <a
              className="button is-small remove"
              onClick={[Function]}
            >
              remove
            </a>
          </td>
        </tr>
        <tr
          className="fieldrow"
          key="dummy2 fielddummy value"
        >
          <td>
            dummy2 field
          </td>
          <td>
            dummy value
          </td>
          <td>
            <a
              className="button is-small remove"
              onClick={[Function]}
            >
              remove
            </a>
          </td>
        </tr>
        <tr
          className="fieldrow"
          key="dummy3 fielddummy value"
        >
          <td>
            dummy3 field
          </td>
          <td>
            dummy value
          </td>
          <td>
            <a
              className="button is-small remove"
              onClick={[Function]}
            >
              remove
            </a>
          </td>
        </tr>
        <tr
          className="fieldrow"
          key="dummy4 fielddummy value"
        >
          <td>
            dummy4 field
          </td>
          <td>
            dummy value
          </td>
          <td>
            <a
              className="button is-small remove"
              onClick={[Function]}
            >
              remove
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DarwinCoreFields>
`;
